test/extended/router/gatewayapicontroller.go

```diff
@@ -370,6 +371,54 @@ func buildGateway(name, namespace, gcname, fromNs, domain string) *gatewayapiv1.
 	}
 }
 
+// assertDNSRecordStatus polls until the DNSRecord's status in the default operand namespace is True.
+func assertDNSRecordStatus(oc *exutil.CLI, gatewayName, domain, lbAddress string) []error {
+	gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+	gateway, errGwStatus := gwapiClient.GatewayV1().Gateways("openshift-ingress").Get(context.TODO(), gatewayName, metav1.GetOptions{})
+	if errGwStatus != nil || gateway == nil {
+		e2e.Failf("Unable to assert dns record, no gateway available, error %v", errGwStatus)
+	}
+
+	// find the DNS Record and confirm its zone status is True
+	gatewayDNSRecord := &operatoringressv1.DNSRecord{}
+	waitErr := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 10*time.Minute, false, func(context context.Context) (bool, error) {
+		gatewayDNSRecords, errDNSStatus := oc.DnsRecordClient().IngressV1().DNSRecords("openshift-ingress").List(context, metav1.ListOptions{})
+		if errDNSStatus != nil {
+			e2e.Logf("Failed to get gateway DNS records, retrying...")
+			return false, nil
+		}
+
+		// get the desired DNS records of the given gateway
+		for count, items := range gatewayDNSRecords.Items {
+			if gatewayDNSRecords.Items[count].Labels["gateway.networking.k8s.io/gateway-name"] == gatewayName {
+				gatewayDNSRecord = &items
+			}
+		}
+
+		// checking whether the dnsName and LB address is present
+		if lbAddress != "" {
+			o.Expect(gatewayDNSRecord.Spec.DNSName).To(o.Equal("*.gwapi." + domain + "."))
+			o.Expect(gatewayDNSRecord.Spec.Targets[0]).To(o.Equal(lbAddress))
+		}
+
+		// checking the gateway DNS record status
+		for _, zones := range gatewayDNSRecord.Status.Zones {
+			for _, condition := range zones.Conditions {
+				if condition.Type == "Published" {
+					if condition.Status == "True" {
+						e2e.Logf("The gateway DNS records created and ready to use")
+						return true, nil
+					}
+				}
+			}
+		}
```

Comment:
If there are two zones, and one status is true, the other is false. I think, the return should not be "true, nil" Form my side, the below update looks good.
```
flag := true
for _, zones := range gatewayDNSRecord.Status.Zones {
     if condition.Type != "Published"  || condition.Status != "True" {
         flag = false
         break
    }
}

if flag {
     e2e.Logf("The gateway DNS records created and ready to use")
}
return flag, nil
```

Reply:
thanks Shudi for the catch..